packages/mongo/mongo_common.js

```diff
@@ -133,6 +133,18 @@ export const replaceMongoAtomWithMeteor = function (document) {
     var buffer = document.value(true);
     return new Uint8Array(buffer);
   }
+  
+  // Check for serialized MongoDB.Binary objects (plain objects with binary-like structure)
+  if (document && typeof document === 'object' && 
+      document.sub_type !== undefined && 
+      document.buffer !== undefined && 
+      document.position !== undefined) {
+    // Convert this serialized binary back to Uint8Array
+    if (document.buffer && (document.buffer instanceof Uint8Array || Array.isArray(document.buffer))) {
+      return new Uint8Array(document.buffer);
+    }
+  }
```

Comment:
This seems like a more suitable place for this conversion, but:
1. Why wasn't it necessary before?
2. Shouldn't we check for `sub_type !== 0`, just like the test above?
3. It'd be more in-align with the rest of the codebase to merge these conditions into one.